cogs/timetableService.py

```diff
@@ -0,0 +1,159 @@
+
+        self.logger.info("TimetableService started successfully")
+
+    @commands.slash_command(
+        name="timetable",
+        description=
+        "Sieh dir den Stundenplan für `today`, `tomorrow` oder die nächsten `n` Tage an",
+        guild_ids=[Constants.SERVER_IDS.CUR_SERVER]
+    )
+    @discord.option(
+        "argument",
+        type=SlashCommandOptionType.string,
+        description=
+        f"`today`, `tomorrow`, Oder eine Zahl (1-{MAX_TIMETABLE_RANGE_DAYS}). Standard ist 7",
+        required=False,
+        default="7"
+    )
+    async def timetable(self, ctx: ApplicationContext, argument: str):
+        """Fetch and display the class timetable.
+
+        This command allows users to view the class schedule for different
+        time periods. Users can request the schedule for today, tomorrow, or
+        any number of upcoming days (max. 30).
+        """
+        await ctx.defer()  # Defer response in case the request takes time
+
+        try:
+            # --- Argument logic ---
+            if argument == "?":
+                response = (
+                    "ℹ️ **Verfügbare Befehle:**\n\n"
+                    "`/timetable today`\n"
+                    "→ Zeigt den Stundenplan für **heute**.\n\n"
+                    "`/timetable tomorrow`\n"
+                    "→ Zeigt den Stundenplan für **morgen**.\n\n"
+                    "`/timetable`\n"
+                    "→ Zeigt den Stundenplan für die **nächsten 7 Tage**.\n\n"
+                    "`/timetable <number>`\n"
+                    "→ Zeigt den Stundenplan für die "
+                    f"**nächsten `<number>` Tage** (max. {MAX_TIMETABLE_RANGE_DAYS})."
+                )
+            elif argument.lower() == "today":
+                response = timetableUtils.get_timetable(days=0)
+            elif argument.lower() == "tomorrow":
+                response = timetableUtils.get_timetable(days=1)
+            elif argument.isdigit():
+                days = int(argument)
+                if days <= 0 or days > MAX_TIMETABLE_RANGE_DAYS:
+                    await ctx.respond(
+                        f"❌ Bitte gib eine Zahl zwischen 1 und {MAX_TIMETABLE_RANGE_DAYS} ein."
+                    )
+                    return
+                response = timetableUtils.get_timetable(days=days)
+            else:
+                response = f"❌ Ungültiger Parameter: {argument or ' - '}. Benutze 'today', 'tomorrow', oder eine Zahl (1-{MAX_TIMETABLE_RANGE_DAYS})."
+
+            await self.send_long_message(ctx, response)
+
+        except Exception as e:
+            self.logger.error("Unhandled error in /timetable: %s", e)
+            await ctx.respond(f"❌ Unbehandelter Fehler: {e}")
+
+    @tasks.loop(time=time(hour=6, minute=0, tzinfo=Constants.SYSTIMEZONE))
+    async def send_daily_timetable(self):
+        """Sends daily timetable message, skips weekends & holidays."""
+        await self.bot.wait_until_ready()
```

Comment:
I am not sure if this is already done earlier.

S: If this is needed, rather place it in the init before the start of the `send_daily_timetable` routine.